PowerEditor/src/WinControls/Grid/ShortcutMapper.cpp

```diff
@@ -121,123 +122,176 @@ void ShortcutMapper::initBabyGrid() {
 	_babygrid.setHighlightColorProtectNoFocus(RGB(230,194,190));
 }
 
+generic_string ShortcutMapper::getTextFromCombo(HWND hCombo)
+{
+	TCHAR str[128];
+	::SendMessage(hCombo, WM_GETTEXT, 127, reinterpret_cast<LPARAM>(str));
+	generic_string res(str);
+	return stringToLower(res);
+};
+
+bool ShortcutMapper::isFilterValid(Shortcut sc)
+{
+	bool match;
+	generic_string shortcut_name = stringToLower(generic_string(sc.getName()));
+	match = false;
+	if (_shortcutFilter[0] == TEXT('\0')){
+		match = true;
+	}
+	// test the filter on the shortcut name
+	size_t match_pos = shortcut_name.find(_shortcutFilter);
+	if (match_pos != std::string::npos){
+		match = true;
+	}
+	return match;
+}

```

Comment:
or even better:
```cpp
if (_shortcutFilter.empty()) {
    return true;
}
```

Reply:
otherwise it should have been (the project is build with both UNICODE and _UNICODE defined):
```cpp
if (_shortcutFilter[0] == L'\0') {
    return true;
}
```

Reply:
done

Reply:
Compacted (nit-picking?):
```cpp
bool ShortcutMapper::isFilterValid(Shortcut sc)
{
	if (_shortcutFilter.empty()) 
	{
		return true;
	}
	// does the name match the filter (which is  stored in lower case)
	return (stringToLower(generic_string(sc.getName())).find(_shortcutFilter) != std::string::npos);
}
```
Allthough that last line may be a bit to compact for readability.